Entity Status Effector

stango1234556 opened the pull request:
functions for buffing and debuffing player damage/speed for limited amount of time

had trouble with pulling and things so changes probs needed before merging

stango1234556 committed:
setup functions for buffing and debuffing player damage/speed for limited amount of time

samsonahh reviewed (changes requested):
The Entity base class already has variables for move speed so I would advise you to look over that.

Instead of using strings to classify your buffs, you can use C# enums. Take a look at the [documentation for it.](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum)

Have you tested this? I believe that you can't construct MonoBehaviours using the new keyword and that you would have to use something like [AddComponent](https://docs.unity3d.com/ScriptReference/GameObject.AddComponent.html)

These buffs are great for the Player, but only for the Player. Generalize this for all entities and make it EntityBuff. EntityBurn.cs (possibly considered a debuff) may help you with that.

BenicioLadines committed:
converted PlayerBuff to StatusEffector

Status effectors are scriptable objects so we can make a ton of unique ones in the editor, status effectors are held by the entity base class now also

BenicioLadines committed:
Merge branch 'development' into Sam-PlayerBuff

stango1234556 committed:
Merge branch 'development' into Sam-PlayerBuff

samsonahh committed:
status effects rework p1

samsonahh committed:
status effect overriding, burning refactored

samsonahh committed:
tick, permanent, and duration status effects with example for each one

samsonahh committed:
hitnumbers juice, status effects ready
